FHIR/src/Microsoft.Health.Fhir.Anonymizer.Shared.Core/Extensions/ElementNodeOperationExtensions.cs

```diff
@@ -20,54 +20,41 @@ public static ElementNode Anonymize(this ElementNode node, AnonymizationFhirPath
         {
             AnonymizationVisitor visitor = new AnonymizationVisitor(rules, processors);
             node.Accept(visitor);
-            node.RemoveNullChildren();
+            node.RemoveEmptyNodes();
 
             return node;
         }
 
-        // Remove null children of current node, and return true => current node is null
-        public static bool RemoveNullChildren(this ElementNode node)
+        public static void RemoveEmptyNodes(this ElementNode node)
         {
             if (node == null)
             {
-                return true;
+                return;
             }
 
-            var children = node.Children().CastElementNodes().ToList();
+            var children = node.Children().ToList();
             foreach (var child in children)
             {
-                // Remove child if it is null => return true
-                if (RemoveNullChildren(child))
+                var elementNodeChild = (ElementNode)child;
+
+                // Remove empty nodes recursively
+                RemoveEmptyNodes(elementNodeChild);
+
+                if (IsEmptyNode(elementNodeChild))
                 {
-                    node.Remove(child);
+                    node.Remove(elementNodeChild);
```

Comment:
seems elementNodeChild could not be null here.

Reply:
Yes, `elementNodeChild` is a child of an ElementNode. The children of ElementNode should never been null. In the library, the children names are accessed directly without checking null: https://github.com/FirelyTeam/firely-net-common/blob/fbc8608775ee6c687031214b4b4d64cb47bfa135/src/Hl7.Fhir.ElementModel/DomNode.cs#L22. Besides, in the original `RemoveNullChildren`, the logic is the same.